datafusion/tests/sql.rs

```diff
@@ -594,6 +594,48 @@ async fn select_distinct_simple_4() {
     assert_batches_sorted_eq!(expected, &actual);
 }
 
+#[tokio::test]
+async fn select_distinct_from() {
+    let mut ctx = ExecutionContext::new();
+
+    let sql = "select 
+        1 IS DISTINCT FROM CAST(NULL as INT) as a,
```

Comment:
I recommend we add tests for `NULL IS DISTINCT FROM NULL` and `NULL IS NOT DISTINCT FROM NULL`  as that is the primary area where this function is different than `=` and `!=`

Reply:
Good idea, done